samples/highcharts/series-heatmap/interpolation/demo.js

```diff
@@ -0,0 +1,130 @@
+const
+    h = 25,
+    w = 30,
+    mapData = [];
+
+for (let i = h; i; i--) {
+    mapData.push(Array(w).fill(0));
+}
+const randomFloored = n => ~~(Math.random() * n);
+
+const withinBound = (val, min, max) => val > min && val < max;
+
+const addIfWithin = (y, x, heat) => {
+    if (withinBound(x, -1, w) && withinBound(y, -1, h)) {
+        mapData[y][x] += heat;
+    }
+};
+
+const clusterRegion = (y, x, i, heat) => {
+    const diagonalHeat = Math.max(heat - 2, 0);
+    const adjacentHeat = Math.max(heat - 1, 0);
+    addIfWithin(y + i, x + i, diagonalHeat);
+    addIfWithin(y + i, x, adjacentHeat);
+    addIfWithin(y, x + i, adjacentHeat);
+};
+
+const placeCluster = (x, y, size) => {
+    addIfWithin(y, x, size);
+
+    for (let i = 1; i <= size; i++) {
+        const heat = size - i;
+        clusterRegion(y, x, i, heat);
+        clusterRegion(y, x, -i, heat);
+        addIfWithin(y - i, x + i, Math.max(0, heat - 2));
+        addIfWithin(y + i, x - i, Math.max(0, heat - 2));
+    }
+};
```

Comment:
This may look confusing to users. In the real world the data set would be coming from a CSV or a database or similar. Please use static data in the form of CSV or JSON or similar. If it is so large that it steals focus from the real sample, it can be moved to ta data file or to a `<pre>` section within the `demo.html` file.

Reply:
Alright! I will probably go for a `<pre>` tag. 

One thing though: Is it ok if I use this to generate the random data or is there some other dataset we should use instead?

Reply:
> Is it ok if I use this to generate the random data

Absolutely 👍